### PR TITLE
Project auth - Andrea L & Pauline A

### PR DESCRIPTION
**Pull request by Andrealarsson**
Link to my deployed project:
Frontend:
https://practical-wozniak-d60eff.netlify.app/Login 
Backend: 
https://project-auth-paan.herokuapp.com

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-## The problem
-
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+Express, React, React Router, Redux, MogoDB/Mongoose, Styled Components, Node.
 
 ## View it live
-
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+Frontend:
+https://practical-wozniak-d60eff.netlify.app/Login
```
Comment (julialeonieros) [inline README.md:11]: A tip! Remove "login" from the URL, otherwise you'll get an error (404) when opening the link

File: backend/server.js [hunk 2]
```diff
@@ -1,28 +1,103 @@
...
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const User = mongoose.model('User', {
+  username: {
+    type: String, 
+    required: true,
+    unique: true, 
+    maxlength: 15,
+    minlength: 5
```
Comment (julialeonieros) [inline backend/server.js:17]: NIce that you've implemented validation on the username! 👍

File: backend/server.js [hunk 3]
```diff
@@ -1,28 +1,103 @@
...
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex')
+  }
+})
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
```
Comment (julialeonieros) [inline backend/server.js:32]: Maybe remove these comments?

File: frontend/src/components/TextInput.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import styled from 'styled-components'
```
Comment (julialeonieros) [inline frontend/src/components/TextInput.js:2]: Another tip! Use 'styled-components/macro' to make is easier to read in dev tools

File: frontend/src/pages/Login.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            }})
+            .catch(error => console.log("error"));
+      }
+
+      return (
+      <Wrapper>
+        <Form onSubmit={handleSubmit} >
+          <Title>Welcome</Title>
+            <TextInput value={value.username} type="text" handleChange={handleChange('username')} title="Username" />
+            <TextInput value={value.password} type="password" handleChange={handleChange('password')} title="Password" />
```
Comment (julialeonieros) [inline frontend/src/pages/Login.js:94]: Maybe some linebreaks here to make it more readable 😇

**Review by julialeonieros — COMMENTED** [review]
Nice and clean project! Easy to read code. We like that you have divided up the components and pages in separate files ✌️ Maybe add that this was a pair-programming project in the Read-Me? Just a tip 😊  You can also delete the file 'SubmitButton.js' that's not in use. Good job! 🥳

File: frontend/src/pages/Login.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+import user from '../reducers/user'
+import TextInput from '../components/TextInput'
+import { API_URL } from '../reusable/urls'
+
+const Wrapper = styled.div`
+width: 100%;
+min-height: 100vh;
+display: flex;
+justify-content: center;
+align-items: center;`
```
Comment (KarolCislowski) [inline frontend/src/pages/Login.js:15]: indentations

File: backend/server.js [hunk 7]
```diff
@@ -1,28 +1,103 @@
...
+    minlength: 5
+  },
+  password: {
+    type: String, 
+    required: true
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex')
+  }
```
Comment (KarolCislowski) [inline backend/server.js:26]: nicely constructed model 👍🏻

File: frontend/src/pages/Login.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+              body: JSON.stringify({
+                username: value.username,
+                password: value.password
+              })
+          }
+
+          fetch(API_URL(path), option)
+          .then((res) => res.json())
+          .then(data => {
+            console.log(data)
```
Comment (KarolCislowski) [inline frontend/src/pages/Login.js:79]: Remove your console logs. It is really important!

File: frontend/src/pages/Login.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+            if(data.success) 
+            { 
+              dispatch(user.actions.setUser({userID: data.userID, accessToken: data.accessToken, errors: null}))
+            } else {
+              dispatch(user.actions.setUser({errors: ""}))
+            }})
+            .catch(error => console.log("error"));
```
Comment (KarolCislowski) [inline frontend/src/pages/Login.js:86]: 👎🏻

File: frontend/src/pages/Main.js [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+const Button = styled.button`
+background: white;
+border: none;
+box-shadow: 2px 2px 1px 2px grey;
+border-radius: 5px;
+`
+
+const Main = () => {
+    const history = useHistory();
+    const dispatch = useDispatch();
```
Comment (KarolCislowski) [inline frontend/src/pages/Main.js:31]: Really bad indentation. 0 and 4 spaces- should be 2 in both cases.

File: frontend/src/reducers/user.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+            setUser: (store, action) => {
+                const { userID, accessToken, errors } = action.payload;
+                store.userID = userID
+                store.accessToken = accessToken
+                store.errors = errors
+            }
+        }
+})
+
+export default user;
```
Comment (KarolCislowski) [inline frontend/src/reducers/user.js:20]: You need to store your user in the `localStorage` to prevent logging out after refreshing.

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
It is really nice app but you need to clean it up and persist your user in the local storage to avoid refreshing.

Because your graduation is so soon please let me know on slack (a short note with a link to this PR) when you finish fixing your app.

**Review by KarolCislowski — APPROVED** [review]
Great App 🥇

**Pull request closed without merging**